homeassistant/components/casatunes/media_player.py

```diff
@@ -0,0 +1,273 @@
+                               discovery_info=None):
+    """Set up the CasaTunes platform."""
+    from casatunes import CasaTunes
+
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+
+    ct = CasaTunes(host, port)
+
+    sources = await ct.get_sources()
+    zones = await ct.get_zones()
+
+    zone_devices = []
+    player_devices = []
+
+    if sources is False:
+        _LOGGER.error("Sources response is incorrect")
+        return False
+
+    if zones is False:
+        _LOGGER.error("Zones response is incorrect")
+        return False
+
+    for zone in zones:
+        try:
+            zone_devices.append(CasaTunesZone(zone))
+        except TypeError:
+            _LOGGER.error("Unable to create a zone")
+            raise PlatformNotReady
+
+    for source in sources:
+        try:
+            player_devices.append(CasaTunesPlayer(source))
+        except TypeError:
+            _LOGGER.error("Unable to create a player")
+            raise PlatformNotReady
+
+    async_add_entities(zone_devices)
+    async_add_entities(player_devices)
+
+
+class CasaTunesPlayer(MediaPlayerDevice):
+    """Representation of a CasaTunes player."""
+
+    def __init__(self, player):
+        """Initialize the CasaTunes player."""
+        self._player = player
+
+    async def async_update(self):
+        """Get the latest details."""
+        await self._player.update()
+
+    @property
+    def media_content_type(self):
+        """Content type of current playing media."""
+        return MEDIA_TYPE_MUSIC
+
+    @property
+    def media_content_id(self):
+        """Return the content ID of current playing media."""
+        return self._player.content_id
+
+    @property
+    def media_title(self):
+        """Title of current playing media."""
+        return self._player.title
+
+    @property
+    def media_artist(self):
+        """Artist of current playing media, music track only."""
+        return self._player.artist
+
+    @property
+    def media_album_name(self):
+        """Album name of current playing media, music track only."""
+        return self._player.album
+
+    @property
+    def media_image_url(self):
+        """Image url of current playing media."""
+        return self._player.image_url
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._player.name
+
+    @property
+    def media_duration(self):
+        """Return the duration of current playing media in seconds."""
+        return self._player.duration
+
+    @property
+    def media_track(self):
+        """Return the track number of current media (Music track only)."""
+        return self._player.track
+
+    @property
+    def shuffle(self):
+        """Boolean if shuffle is enabled."""
+        return self._player.shuffle
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        status = self._player.status
+
+        if status == 0:
```

Comment:
Create a mapping dictionary constant instead, i.e.
```python
STATUS_TO_STATE = {
    0: STATE_IDLE,
    1: STATE_PAUSED
    2: STATE_PLAYING,
    3: STATE_ON
}
```

Then here you can do `return STATUS_TO_STATE[self._player.status]`